cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go

```diff
@@ -226,7 +240,20 @@ func (agentPool *ContainerServiceAgentPool) setACSNodeCount(count int) error {
 		klog.Errorf("Failed to update ACS cluster (%q): %v", agentPool.clusterName, err)
 		return err
 	}
-	return future.WaitForCompletionRef(updateCtx, acsClient.Client)
+
+	err = future.WaitForCompletionRef(updateCtx, acsClient.Client)
+	resp := future.Response()
+	if isSuccessHTTPResponse(resp) {
```

Comment:
the code is a little duplicate in many places, what about write`isSuccessHTTPResponse` as
```
func isSuccessHTTPResponse(resp *http.Response) (bool, err error)
```
if failed, then return error message directly.

Reply:
Good catch, would fix in the new commit.